### PR TITLE
porting API services to used Index Service v2

### PR DESCRIPTION
**Pull request by gaboraranyossy-da**
See title.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala [hunk 1]
```diff
@@ -22,27 +22,31 @@ package object v2 {
...
         templateId: Ref.Identifier,
         argument: Value.VersionedValue[Value.AbsoluteContractId],
         // TODO(JM,SM): understand witnessing parties
-        stakeholders: List[Ref.Party],
+        stakeholders: Set[Ref.Party],
     ) extends AcsUpdateEvent
 
     final case class Archive(
         transactionId: TransactionId,
-        nodeId: AbsoluteNodeId,
```
Comment (gerolf-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: I guess we should remove `AbsoluteNodeId` completely?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala [hunk 2]
```diff
@@ -178,6 +182,8 @@ class SandboxLedgerBackend(ledger: Ledger)(implicit mat: Materializer)
       )
   }
 
+  //def getTransactionById(transactionId: TransactionId): Future[Option[AcceptedTransaction]]
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala]: ?

File: ledger/api-server-damlonx/BUILD.bazel [hunk 3]
```diff
@@ -41,5 +41,6 @@ da_scala_library(
         "//ledger/ledger-api-scala-logging",
         "//ledger/participant-state",
         "//ledger/participant-state-index",
+        "//ledger/participant-state-index:participant-state-index-v2",
```
Comment (gerolf-da) [inline ledger/api-server-damlonx/BUILD.bazel]: daml-on-x shouldn't use the v2 index api.

Comment (gaboraranyossy-da) [reply ledger/api-server-damlonx/BUILD.bazel]: Only `v2.Identityservice` is needed. Could we keep it?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala (hunk 2 above)
Comment (gaboraranyossy-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala]: thanks for noticing, will delete it.

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala (hunk 1 above)
Comment (gaboraranyossy-da) [reply ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: what do you mean? It's not used anymore in this file, is it?

File: ledger/api-server-damlonx/BUILD.bazel (hunk 3 above)
Comment (gerolf-da) [reply ledger/api-server-damlonx/BUILD.bazel]: nothing of v2 participant-state-index should be used in damlonx stuff.

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: I meant to replace it with `EventId` and removing it completely from the codebase. I introduced it in the v2 index-api PR.

Comment (gaboraranyossy-da) [reply ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: it is replaced with `domain.EventId`

File: ledger/api-server-damlonx/BUILD.bazel (hunk 3 above)
Comment (gaboraranyossy-da) [reply ledger/api-server-damlonx/BUILD.bazel]: I massaged the code so we can get it away without it.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala [hunk 4]
```diff
@@ -123,12 +125,13 @@ class SandboxLedgerBackend(ledger: Ledger)(implicit mat: Materializer)
...
 
   private def toUpdateEvent(
-      id: Value.AbsoluteContractId,
+      cId: Value.AbsoluteContractId,
       ac: ActiveContracts.ActiveContract): AcsUpdateEvent.Create =
     AcsUpdateEvent.Create(
       // we use absolute contract ids as event ids throughout the sandbox
-      id.coid,
-      id,
+      domain.TransactionId(cId.coid), //TODO: is this correct?
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala]: `ActiveContracts.ActiveContract` has a field `transactionId`.

Comment (gaboraranyossy-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala]: indeed, fixed

**Pull request merged**